internal/providers/github/properties/release.go

```diff
@@ -0,0 +1,154 @@
+// SPDX-FileCopyrightText: Copyright 2024 The Minder Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package properties
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	go_github "github.com/google/go-github/v63/github"
+
+	"github.com/mindersec/minder/internal/entities/properties"
+	minderv1 "github.com/mindersec/minder/pkg/api/protobuf/go/minder/v1"
+	v1 "github.com/mindersec/minder/pkg/providers/v1"
+)
+
+// Release Properties
+const (
+	// ReleasePropertyOwner represents the github owner
+	ReleasePropertyOwner = "github/owner"
+	// ReleasePropertyRepo represents the github repo
+	ReleasePropertyRepo = "github/repo"
+	// ReleasePropertyTag represents the github release tag name.
+	ReleasePropertyTag = "github/tag"
+	// ReleasePropertyBranch represents the github release branch
+	ReleasePropertyBranch = "github/branch"
+)
+
+// ReleaseFetcher is a property fetcher for releases
+type ReleaseFetcher struct {
+	propertyFetcherBase
+}
+
+// NewReleaseFetcher creates a new ReleaseFetcher
+func NewReleaseFetcher() *ReleaseFetcher {
+	return &ReleaseFetcher{
+		propertyFetcherBase: propertyFetcherBase{
+			propertyOrigins: []propertyOrigin{
+				{
+					keys: []string{
+						// general entity
+						properties.PropertyName,
+						properties.PropertyUpstreamID,
+						// general release
+						ReleasePropertyTag,
+						ReleasePropertyBranch,
+					},
+					wrapper: getReleaseWrapper,
+				},
+			},
+			operationalProperties: []string{},
+		},
+	}
+}
+
+// GetName returns the name of the release
+func (_ *ReleaseFetcher) GetName(props *properties.Properties) (string, error) {
+	owner := props.GetProperty(ReleasePropertyOwner).GetString()
+	repo, err := props.GetProperty(ReleasePropertyRepo).AsString()
+	if err != nil {
+		return "", fmt.Errorf("failed to get repo name: %w", err)
+	}
+
+	tag, err := props.GetProperty(ReleasePropertyTag).AsString()
+	if err != nil {
+		return "", fmt.Errorf("failed to get tag name: %w", err)
+	}
+
+	return getReleaseNameFromParams(owner, repo, tag), nil
+}
+
+func getReleaseNameFromParams(owner, repo, tag string) string {
+	if owner == "" {
```

Comment:
This seems like a weird case -- does it actually happen, or could we put in a complaint?

Reply:
Let's make this an error and/or log at warning/error -- I think it's valid input to the GitHub API, but we shouldn't be getting _output_ in this non-normalized form.